File: 1668.maximum-repeating-substring.cs
// @leet start
public class Solution {
    public int MaxRepeating(string sequence, string word) {
      var max = 0;
      for(int i = 0; i < sequence.Length; i++) {
        if (i + word.Length > sequence.Length) break;
        var count = 0;
        var j = i;
        while(sequence.Substring(j, word.Length) == word) {
          count++;
          j += word.Length;
          if (j + word.Length > sequence.Length) break;
        }
        if (count > max) max = count;
      }
      return max;
    }
}
// @leet end

File: 1668.maximum-repeating-substring.kt
// @leet start
class Solution {
    fun maxRepeating(sequence: String, word: String): Int {
      var max = 0
      for (i in sequence.indices) {
        if (sequence[i] != word[0]) continue
        if (i + word.length > sequence.length) break
        var j = i
        var count = 0
        while (sequence.substring(j, j + word.length) == word) {
          count++
          if (count > max) max = count
          j += word.length
          if (j + word.length > sequence.length) break
        }
      }
      return max
    }
}
// @leet end
